dynamo-event-to-datacite-request/src/test/java/no/unit/nva/events/models/doi/dto/PublicationTypeTest.java

```diff
@@ -0,0 +1,68 @@
+package no.unit.nva.events.models.doi.dto;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.containsString;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.is;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.CsvSource;
+
+class PublicationTypeTest {
+
+    public static final String UNDEFINED = "UNKNOWN";
+    public static final String NO_ENUM_CONSTANT_FOR = "No enum constant for";
+
+    @ParameterizedTest
+    @CsvSource({
+        "JOURNAL_ARTICLE,JournalArticle",
+        "JOURNAL_LETTER,JournalLetter",
+        "JOURNAL_LEADER,JournalLeader",
+        "JOURNAL_REVIEW,JournalReview",
+        "JOURNAL_SHORT_COMMUNICATION,JournalShortCommunication",
+        "BOOK_MONOGRAPH,BookMonograph",
+        "BOOK_ANTHOLOGY,BookAnthology",
+        "DEGREE_BACHELOR,DegreeBachelor",
+        "DEGREE_MASTER,DegreeMaster",
+        "DEGREE_PHD,DegreePhd",
+        "REPORT,Report",
+        "REPORT_POLICY,ReportPolicy",
+        "REPORT_RESEARCH,ReportResearch",
+        "REPORT_WORKING_PAPER,ReportWorkingPaper",
+        "CHAPTER_ARTICLE,ChapterArticle"
+    })
+    void testToString(PublicationType enumValue, String expectedToString) {
+        assertThat(enumValue.toString(), is(equalTo(expectedToString)));
+    }
+
+    @ParameterizedTest
+    @CsvSource({
+        "JOURNAL_ARTICLE,JournalArticle",
+        "JOURNAL_LETTER,JournalLetter",
+        "JOURNAL_LEADER,JournalLeader",
+        "JOURNAL_REVIEW,JournalReview",
+        "JOURNAL_SHORT_COMMUNICATION,JournalShortCommunication",
+        "BOOK_MONOGRAPH,BookMonograph",
+        "BOOK_ANTHOLOGY,BookAnthology",
+        "DEGREE_BACHELOR,DegreeBachelor",
+        "DEGREE_MASTER,DegreeMaster",
+        "DEGREE_PHD,DegreePhd",
+        "REPORT,Report",
+        "REPORT_POLICY,ReportPolicy",
+        "REPORT_RESEARCH,ReportResearch",
+        "REPORT_WORKING_PAPER,ReportWorkingPaper",
+        "CHAPTER_ARTICLE,ChapterArticle"
+    })
+    void testFindByName(PublicationType expectedEnum, String findByNameValue) {
+        assertThat(PublicationType.findByName(findByNameValue), is(equalTo(expectedEnum)));
+    }
+
+    @Test
+    void testWhenUnknownTypeFindByNameThenThrowsIllegalArgumentException() {
```

Comment:
You can reduce the name of the method by removing uneccessary words. "Test" is unecessary because we already know it is a test. "Then" can be removed by following the pattern `findByNameThrowsExceptionWhenUnknownTypeInserted`